prac_05/emails.py

```diff
@@ -0,0 +1,38 @@
+"""
+CP1404/CP5632 Practical 05
+Emails
+Estimate: 25 minutes
+Actual:  minutes
+"""
+
+
+def main():
+    """Main program to store users emails and names in a dictionary"""
+    emails_to_names = {}
+    email = input("Email: ").lower()
+    while email != "":
+        name = extract_name(email)
+
+        name_check = input(f"Is your name {name}? (Y/n) ").lower()
+
+        if name_check != "" and name_check != "y":
+            name = input("Name: ")
+        emails_to_names[email] = name
+
+        email = input("Email: ").lower()
+
+    print()
+    for email, name in emails_to_names.items():
+        print(f"{name} ({email})")
+
+
+def extract_name(email):
+    """Extract name from a given email"""
+    name = email.split("@")
+    name = name[0]
+    name = name.replace(".", " ")
+
```

Comment:
Your code can't handle emails with numbers or any other special characters
could include code like:
characters_to_remove = ".,!?'*$#1234567890"
 name = " ".join(username_parts)
 name = "".join(char for char in name if char not in characters_to_remove)

Reply:
The sample output didn't seem to account for numbers, so neither did I. 
![image](https://github.com/sachhm/cp1404practicals/assets/39152811/78d52183-df06-4bee-b59f-3104bbd5696c)